src/common/store/background-epics.ts

```diff
@@ -16,15 +30,53 @@ interface FluxStandardAction {
 export const locateEpic: Epic<FluxStandardAction> = action$ => action$.pipe(
   ofType('API_READ'),
   filter(action => action.payload.endpoint === endpoints.ANNOTATIONS),
-  mergeMap(async (action) => {
-    const tabId = retrieveActionTab(action);
-    const locationData = await tabLocateAnnotations(tabId, action.payload.data);
-    syncBadgeWithAnnotations(locationData, tabId);
-    const newAction = locateAnnotations(locationData);
-    return syncTabMark(action, newAction);
-  }),
+  mergeMap(action => defer(
+    async () => {
+      const tabId = retrieveActionTab(action);
+      const locationData = await tabLocateAnnotations(tabId, action.payload.data);
+      syncBadgeWithAnnotations(locationData, tabId);
+      const newAction = locateAnnotations(locationData);
+      return syncTabMark(action, newAction);
+    },
+  )),
+);
+
+export const processAuthenticationEpic: Epic<FluxStandardAction, any, IState> = (action$, state$) => (
+  action$.pipe(
+    ofType(SET_AUTH_PROC_STAGE),
+    filter(() => !selectUser(state$.value)),
+    filter(action => action.payload.stage === AuthProcStages.initiated),
+    mergeMap( (action): Observable<FluxStandardAction> =>
+      from(authenticate(action.payload.provider)).pipe(
+        mergeMap( loginData =>
+          loginData ? of(
+            completeAuthProc(),
+            userDataNew(loginData),
+          ) : of(
+            cancelAuthProc('Anulowano autoryzację'),
+          ),
+        ),
+        catchError((err) => {
+          Sentry.captureException(err);
+          return of(failAuthProc('Wystąpił bład podczas logowania'));
+        }),
+        map(newAction => syncTabMark(action, newAction)),
+      ),
+    ),
+  )
+);
+
+export const propagateAuthenticationDataEpic: Epic<FluxStandardAction, any, IState> = action$ => (
+  action$.pipe(
+    filter(action => [USER_DATA_NEW, USER_ACCESS_TOKEN_REFRESHED, USER_DATA_CLEARED].includes(action.type)),
```

Comment:
Actually that's quite a nice way to declare relationship between these actions and login info propagation to dashboard :)